api/src/services/history-publish-service.ts

```diff
@@ -247,4 +247,82 @@ export class HistoryPublishService extends DBService {
   async deleteSurveyReportAttachmentPublishRecord(surveyReportAttachmentId: number): Promise<void> {
     return this.historyRepository.deleteSurveyReportAttachmentPublishRecord(surveyReportAttachmentId);
   }
+
+  async hasUnpublishedSurveyAttachments(surveyId: number): Promise<boolean> {
```

Comment:
nit: maybe just add some jsdoc to these few functions, explain what a `true` vs `false` means

Reply:
added